src/main/java/actionlogger/writers/JsonWriter.java

```diff
@@ -40,52 +40,66 @@ public JsonWriter(Gson gson, Client client, ExecutorService executor) {
         this.restartFile();
     }
 
-    public void restartFile() {
-        if (this.fh != null) {
-            this.close();
-        }
+    public CompletableFuture<Map.Entry<Path, Path>> restartFile() {
+        CompletableFuture<Map.Entry<Path, Path>> future = new CompletableFuture<>();
+        executor.execute(() -> {
+            var oldFh = this.fh;
+            var oldPath = this.path;
+            var path = dir.toPath().resolve(String.format("%d-logs.txt", System.currentTimeMillis()));
+            try {
+                this.fh = Files.newBufferedWriter(path);
+                this.path = path;
+
+                if (oldFh != null) {
+                    try {
+                        oldFh.close();
+                    } catch (IOException e) {
+                        log.warn("Failed to close old writer for {}", oldPath, e);
```

Comment:
Is this missing a `{}` or is this fine?

Reply:
It's fine; `Throwable` as last arg has special treatment https://www.slf4j.org/faq.html#paramException